[update ] Promscale CLI reference (#1132)
timescale/docs: promscale/cli.md

# Promscale command line interface (CLI) reference

This document gives you information about the configuration flags and arguments
supported by the Promscale Connector. You can also find information on flags
with `promscale_<version> -help`.

> **Note:** Flags can be set as environment variables by converting lowercase to

Reviewer: Can we add this to a special section i.e. by using `<highlight type="">`

Author: done.

| db.uri | string | | TimescaleDB/Vanilla PostgresSQL URI. Example:`postgres://postgres:password@localhost:5432/timescale?sslmode=require` |
| db.user | string | postgres | TimescaleDB/Vanilla Postgres user. |

## Telemetry flags (for telemetry generated by the Promscale connector)

| Flag | Type | Default | Description |
|------|:-----:|:-------:|:-----------|
| telemetry.log.format | string | logfmt | Log format to use from [ "logfmt", "json" ]. |
| telemetry.log.level | string | debug | Log level to use from [ "error", "warn", "info", "debug" ]. |
| telemetry.log.throughput-report-interval | duration | 0 second | Duration interval at which throughput should be reported. Setting duration to `0` will disable reporting throughput, otherwise, an interval with unit must be provided, e.g. `10s` or `3m`. |
| telemetry.trace.otel-endpoint | string | "" (empty) | OpenTelemetry tracing collector GRPC URL endpoint to send telemetry to otel-collector:4317 |

> | telemetry.trace.otel-tls-cert-file | string | "" (empty) | TLS Certificate file used for client authentication against the OTEL tracing collector GRPC endpoint, leave blank to disable TLS. |
> | telemetry.trace.otel-tls-key-file | string | "" (empty) | TLS Key file for client authentication against the OTEL tracing collector GRPC endpoint, leave blank to disable TLS. |

Reviewer 2: ```suggestion
| telemetry.trace.otel-tls-cert-file | string | "" (empty) | TLS Certificate file used for client authentication against the OTEL tracing collector GRPC endpoint. Leave blank to disable TLS. |
| telemetry.trace.otel-tls-key-file | string | "" (empty) | TLS Key file for client authentication against the OTEL tracing collector GRPC endpoint. Leave blank to disable TLS. |
```

> | telemetry.trace.jaeger-endpoint | string | "" (empty) | Jaeger tracing collector thrift HTTP URL endpoint to send telemetry to (e.g. https://jaeger-collector:14268/api/traces). |

Reviewer 2: ```suggestion
| telemetry.trace.jaeger-endpoint | string | "" (empty) | Jaeger tracing collector thrift HTTP URL endpoint to send telemetry to (for example: https://jaeger-collector:14268/api/traces). |
```

| telemetry.trace.sample-ratio | float | 1.0 | Trace sampling ratio, amount of spans to send to collector. Valid values from 0.0 (none) to 1.0 (all). |

## Metrics specific flags flags

| Flag | Type | Default | Description |
|------|:-----:|:-------:|:-----------|
| metrics.async-acks | boolean | false | Acknowledge asynchronous inserts. If this is true, the inserter will not wait after insertion of metric data in the database. This increases throughput at the cost of a small chance of data loss.|
| metrics.cache.exemplar.size | unsigned-integer | 10000 | Maximum number of exemplar metrics key-position to cache. It has one-to-one mapping with number of metrics that have exemplar, as key positions are saved per metric basis. |
| metrics.cache.labels.size | unsigned-integer | 10000 | Maximum number of labels to cache.|
| metrics.cache.metrics.size | unsigned-integer | 10000 | Maximum number of metric names to cache. |
| metrics.cache.series.initial-size | unsigned-integer | 250000 | Initial number of elements in the series cache. |

> | metrics.cache.series.max-bytes | unsigned-integer or percentage | 50% | Target for amount of memory to use for the series cache. Specified in bytes or as a percentage of the memory-target. For example, 50%.|

Reviewer 2: ```suggestion
| metrics.cache.series.max-bytes | unsigned-integer or percentage | 50% | Target for amount of memory to use for the series cache. Specified in bytes or as a percentage of `memory-target`. For example, 50%.|
```

> | metrics.high-availability | boolean | false | Enable external_labels based HA.|

Reviewer 2: ```suggestion
| metrics.high-availability | boolean | false | Enable external_labels-based HA.|
```

| metrics.ignore-samples-written-to-compressed-chunks | boolean | false | Ignore/drop samples that are being written to compressed chunks. Setting this to false allows Promscale to ingest older data by decompressing chunks that were earlier compressed. However, setting this to true will save your resources that may be required during decompression. |
| metrics.multi-tenancy | boolean | false | Use multi-tenancy mode in Promscale. |
| metrics.multi-tenancy.allow-non-tenants | boolean | false | Allow Promscale to ingest/query all tenants as well as non-tenants. By setting this to true, Promscale will ingest data from non multi-tenant Prometheus instances as well. If this is false, only multi-tenants (tenants listed in 'multi-tenancy-valid-tenants') are allowed for ingesting and querying data. |

> | metrics.multi-tenancy.valid-tenants | string | allow-all | Sets valid tenants that are allowed to be ingested/queried from Promscale. This can be set as: 'allow-all' (default) or a comma separated tenant names. 'allow-all' makes Promscale ingest or query any tenant from itself. A comma separated list will indicate only those tenants that are authorized for operations from Promscale. |

Reviewer 2: ```suggestion
| metrics.multi-tenancy.valid-tenants | string | allow-all | Sets valid tenants that are allowed to be ingested/queried from Promscale. This can be set as 'allow-all' (default), or as comma separated tenant names. 'allow-all' makes Promscale ingest or query any tenant from itself. A comma separated list indicates only those tenants that are authorized for operations from Promscale. |
```

| metrics.promql.default-subquery-step-interval | duration | 1 minute | Default step interval to be used for PromQL subquery evaluation. This value is used if the subquery does not specify the step value explicitly. Example: <metric_name>[30m:]. Note: in Prometheus this setting is set by the evaluation_interval option. |
| metrics.promql.lookback-delta | duration | 5 minute | The maximum look-back duration for retrieving metrics during expression evaluations and federation. |

> | metrics.promql.max-points-per-ts | integer64 | 11000 | Maximum number of points per time-series in a query-range request. This calculation is an estimation, that happens as (start - end)/step where start and end are the 'start' and 'end' timestamps of the query_range. |

Reviewer 2: ```suggestion
| metrics.promql.max-points-per-ts | integer64 | 11000 | Maximum number of points per time-series in a query-range request. This calculation is an estimation, equal to (start - end)/step where start and end are the 'start' and 'end' timestamps of the query_range. |
```

| metrics.promql.max-samples | integer64 | 50000000 | Maximum number of samples a single query can load into memory. Note that queries will fail if they try to load more samples than this into memory, so this also limits the number of samples a query can return. |
| metrics.promql.query-timeout | duration | 2 minutes | Maximum time a query may take before being aborted. This option sets both the default and maximum value of the 'timeout' parameter in '/api/v1/query.*' endpoints. |

## Recording and Alerting rules flags 
| Flag | Type | Default | Description |
|------|:-----:|:-------:|:-----------|
| metrics.alertmanager.notification-queue-capacity | integer | 10000 | The capacity of the queue for pending Alertmanager notifications. |

> | metrics.rules.alert.for-grace-period | duration | 10 minutes | Minimum duration between alert and restored "for" state. This is maintained only for alerts with configured "for" time greater than grace period.|

Reviewer 2: ```suggestion
| metrics.rules.alert.for-grace-period | duration | 10 minutes | Minimum duration between alert and restored "for" state. This is maintained only for alerts with configured "for" time greater than the grace period.|
```

| metrics.rules.alert.for-outage-tolerance | duration | 1 hour | Max time to tolerate Promscale outage for restoring "for" state of alert. |
| metrics.rules.alert.resend-delay | duration |  1 minute  | Minimum amount of time to wait before resending an alert to Alertmanager. |

> | metrics.rules.config-file | string | "" | Path to configuration file in Prometheus-format, containing rule_files and optional `alerting`, `global` fields. For more details, see https://prometheus.io/docs/prometheus/latest/configuration/configuration/. Note: If this is flag or `rule_files` is empty, Promscale rule-manager will not start. If `alertmanagers` is empty, alerting will not be initialized. |

Reviewer 2: ```suggestion
| metrics.rules.config-file | string | "" | Path to configuration file in Prometheus-format, containing rule_files and optional `alerting` and `global` fields. For more details, see https://prometheus.io/docs/prometheus/latest/configuration/configuration/. Note: If this flag is missing or `rule_files` is empty, Promscale rule-manager does not start. If `alertmanagers` is empty, alerting is not initialized. |
```
